techdocs/script/test_operations.py

```diff
@@ -1,55 +1,163 @@
+        ),
         file,
         Mock(
             config=Mock(
+                project="promil",
                 source=rule_source,
                 destination=rule_destination,
             )
         ),
     )
     assert copy_operation.source_abs == expected_source
-    assert (
-        copy_operation.destination_abs
-        == expected_destination
+    assert copy_operation.destination_abs == expected_destination
+
+
+def test_index_load():
+    fs = MockFilesystem(
+        {
+            "/foo/index/Promil-platform-foo/42af564a885e1f38be3f411de2584efc3462bba68e9b5ea6dc39364b061d0a8f": json.dumps(
+                {
+                    "file": "heheszek",
+                    "repo": "Promil-platform-foo",
+                }
+            )
+        }
     )
+
+    items = IndexLoader.load("/foo/index", fs).items
+
+    assert items[0].file == "heheszek"
+    assert items[0].repo == "Promil-platform-foo"
+    assert len(items) == 1
+
+
+def test_index_save():
+    fs = MockFilesystem({})
+
+    index = Index(
+        (
+            IndexItem("heheszek", "Promil-platform-foo"),
+            IndexItem("foo/bar", "Promil-platform-foo"),
+            IndexItem("baz/huehue", "Promil-platform-bar"),
+        )
+    )
+    IndexLoader.save(index, "/foo/index", fs)
+
+    assert list(sorted(fs.scan("/foo/index", ".*"))) == list(
+        sorted(
+            [
+                "Promil-platform-foo/42af564a885e1f38be3f411de2584efc3462bba68e9b5ea6dc39364b061d0a8f",
+                "Promil-platform-foo/cc5d46bdb4991c6eae3eb739c9c8a7a46fe9654fab79c47b4fe48383b5b25e1c",
+                "Promil-platform-bar/4db8a688c5803846883870d0e15d84c33fb492ad5d8b67a9b7199d3eeeaa1907",
+            ]
+        )
+    )
+
+    assert json.loads(
+        fs.read_string(
+            "/foo/index/Promil-platform-foo/42af564a885e1f38be3f411de2584efc3462bba68e9b5ea6dc39364b061d0a8f"
+        )
+    ) == {
+        "file": "heheszek",
+        "repo": "Promil-platform-foo",
+    }
+
+
+def test_delete():
+    fs = MockFilesystem(
+        {
+            "/tmp/Promil/a-file": "a-file-content",
+            "/tmp/dst/a-file": "a-file-content",
+            "/tmp/dst/a-file-that-does-not-exist-anymore": "blabla",
+        }
+    )
+    mock_detector = Mock(
+        detect=Mock(
+            return_value=[
+                FilesystemOperation(
+                    source_abs="/tmp/Promil/a-file",
+                    destination_abs="/tmp/dst/a-file",
+                    type=FilesystemOperation.TYPE_COPY,
+                )
+            ]
+        )
+    )
+    index = Index(
+        (
+            IndexItem("a-file", "Promil"),
+            IndexItem("a-file-that-does-not-exist-anymore", "Promil"),
+        )
+    )
+    detector = DeleteDetector(
+        "Promil",
+        index,
+        "/tmp/Promil",
+        "/tmp/dst",
+        mock_detector,
+    )
+
+    operations = detector.detect(fs)
```

Comment:
For `CopyDetector` you didn't test out `detect()` but went with `_create_operation`:
https://github.com/PiwikPRO/actions/blob/dec3734ec1766752b7ec644f9a40db23042bb99f/techdocs/script/test_operations.py#L52-L53

There is some minor logic in detect as well so maybe it's worth to align those detector tests:
https://github.com/PiwikPRO/actions/blob/dec3734ec1766752b7ec644f9a40db23042bb99f/techdocs/script/operations.py#L43-L48

Reply:
Good point. I didn't have MockFilesystem when I started writing those tests, that's the reason.